web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,196 @@
+
+### Missing Value Handling
+
+Missing value handling is a tricky area as developers are split between pd.NA semantics versus np.nan, and the transition path from one to the other is not always clear. This PDEP does not aim to "solve" that issue per se (for that discussion, please refer to PDEP-16), but aims to provide a go-forward path that strikes a reasonable balance between backwards compatibility and a consistent missing value approach in the future.
+
+This PDEP proposes that the default missing value for logical types is ``pd.NA``. The reasoning is two-fold:
+
+  1. We are in many cases re-using extension types as logical types, which mostly use pd.NA (StrDtype and datetimes are the exception)
+  2. For new logical types that have nothing to do with NumPy, using np.nan as a missing value marker is an odd fit
+
+However, to help with backwards compatibility for users that heavily rely on the semantics of ``np.nan`` or ``pd.NaT``, an option of ``pd.na_marker = "legacy"`` can be set. This would mean that the missing value indicator for logical types would be:
+
+| Logical Type      | Default Missing Value | Legacy Missing Value |
+| pd.BooleanDtype() | pd.NA                 | np.nan               |
+| pd.IntXXType()    | pd.NA                 | np.nan               |
+| pd.FloatXXType()  | pd.NA                 | np.nan               |
+| pd.StringDtype()  | pd.NA                 | np.nan               |
+| pd.DatetimeType() | pd.NA                 | pd.NaT               |
+
+However, all data types for which there is no legacy NumPy-backed equivalent will continue to use ``pd.NA``, even in "legacy" mode. Legacy is provided only for backwards compatibility, but ``pd.NA`` usage is encouraged going forward to give users one exclusive missing value indicator and better align with the goals of PDEP-16.
+
+### Transitioning from Current Constructors
+
+To maintain a consistent path forward, _all_ constructors with the implementation of this PDEP are expected to map to the logical types. This means that providing ``np.int64`` as the data type argument makes no guarantee that you actually get a NumPy managed storage buffer; pandas reserves the right to optimize as it sees fit and may decide instead to use PyArrow.
+
+The theory behind this is that the majority of users are not expecting anything particular from NumPy to happen when they say ``dtype=np.int64``. The expectation is that a user just wants _integer_ data, and the ``np.int64`` specification owes to the legacy of pandas' evolution.
```

Comment:
IIUC the idea here is that if a user writes dtype=np.int64 they might not really "mean it" so we'll ignore the "np" part of it, but if they write dtype=pd.Int64Dtype(backend="numpy") we _will_ respect it?

Reply:
Yea, at least as far as the initial storage goes. I am operating under the (hopefully valid) assumption that the need to be really specific about the storage is an exception